test/Shared/StrongNameTests.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma warning disable IDE0005
+using Xunit;
+#pragma warning restore IDE0005
+
+namespace OpenTelemetry.Tests;
+
+// The tests can only be strong named if they only depend on assemblies that are strong named,
+// therefore if the tests are strong named then the libraries we ship are strong named.
+
+public static class StrongNameTests
+{
+    [Fact]
+    public static void Tests_Are_Strong_Named()
+    {
+        // Arrange
+        var assembly = typeof(StrongNameTests).Assembly;
+        var name = assembly.GetName();
+
+        if (name.Name == "OpenTelemetry.Instrumentation.Hangfire.Tests")
+        {
+            // Hangfire tests are not strong named (SignAssembly=false), so skip
+            return;
+        }
```

Comment:
if Hangfire later adds strong naming, is there any risk of getting false positives from this test?

I wonder if it would be more resilient to have a property in individual projects to opt ouf of this test. Something like: `<SkipStrongNameValidation>true</SkipStrongNameValidation>`